### ISSUE TITLE
Remove index UI selectors.

### ISSUE THREAD

**Issue by eeejay** [mozilla-mobile/fenix#24490]
### Firebase Test Run:
[Provide a Firebase test run report link here showcasing the problem](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/5849132562068109450/details?stepId=bs.4a1bac1658a9f03&testCaseId=1)

We plan to enable a new accessibility cache in GeckoView, it was merged once and then reverted, partially because of Fenix UI test failures. Some of these failures are due to the use of the `index()` selector that selects objects in a certain index in relation to their siblings. The UI tree changes slightly when cache is enabled and those index values are no longer correct.

Instead of updating them, I think they should be removed because we already have resource ID selectors which are much more specific. Index selectors are redundant.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-19985)
